test/src/basic/fileformats.cpp

```diff
@@ -697,6 +695,47 @@ void describeDOTwithClusters()
 				AssertThat(cluster->nodes.size(), Equals(4));
 			}
 		});
+
+		it("reads assignment statements", []() {
+			const ResourceFile* file = ResourceFile::get(
+				"fileformats/dot/valid/assignments");
+			std::stringstream is{file->data()};
+
+			Graph G;
+			ClusterGraph CG(G);
+			ClusterGraphAttributes CGA(CG);
+
+			const bool readStatus = GraphIO::readDOT(CGA, CG, G, is);
+			AssertThat(readStatus, Equals(true));
+
+			AssertThat(CGA.label(CG.rootCluster()), Equals("wat"));
+		});
+
+		it("parses dir attribute", []() {
+			const ResourceFile* file = ResourceFile::get(
+				"fileformats/dot/valid/arrowtypes");
+			std::stringstream is{file->data()};
+
+			Graph G;
+			GraphAttributes GA(G, GraphAttributes::edgeArrow);
+
+			const bool readStatus = GraphIO::readDOT(GA, G, is);
+			AssertThat(readStatus, Equals(true));
+
+			const std::vector<EdgeArrow> arrowTypes = {
+				EdgeArrow::Both,
+				EdgeArrow::Last,
+				EdgeArrow::First,
+				EdgeArrow::None,
+				EdgeArrow::Undefined
+			};
+
+			auto edge = G.firstEdge();
+			for (auto &arrowType : arrowTypes) {
```

Comment:
For test atomicity, I'd prefer to have the `it()` inside the for-loop. (So you should probably use an `std::vector<std::pair<EdgeArrow, string>> arrowTypes = { {EdgeArrow::Both, "both"}, ... };` and do something like `it("recognizes dir attribute set to " + arrowType.second, [] { ... })` in each iteration.)

Reply:
Btw, an even better idea is to drop that weird vector of pairs. It would instead benefit readability if you move the `it()` part into a lambda, e.g. `auto testDirAttribute = [&](EdgeArrow, string) { ... }` and then do
```c++
    testDirAttribute(EdgeArrow::Both, "both");
    testDirAttribute(EdgeArrow::Last, "last");
    testDirAttribute(EdgeArrow::First, "first");
    testDirAttribute(EdgeArrow::None, "none");
```

Reply:
That's a great idea. Do you think it's better to pass the string into the lambda instead of using the `ogdf::dot::toString(const EdgeArrow)` function?

Reply:
No, I wasn't aware that there is a `toString()` function ;-)

Reply:
I've decided it's better not to use the `dot::toString(const EdgeArrow)` function because we should test for bugs in `toString` itself. Besides, it makes no sense to `#include <ogdf/fileformats/DOT.h>` inside the test file.